src/common-priv.h

```diff
@@ -0,0 +1,77 @@
+/*-
+ * Copyright (c) 2019 Ken Cunningham kencu@macports.org
+ *
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE AUTHOR AND CONTRIBUTORS ``AS IS'' AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED.  IN NO EVENT SHALL THE AUTHOR OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
+ * OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
+ * SUCH DAMAGE.
+ */
+
+#ifndef macports_common_priv_h_
+#define macports_common_priv_h_
+
+/* MP support header */
+#include "MacportsLegacySupport.h"
+
+/* Should be generic enough so that we don't need a global feature macro. */
+
+#include <sys/errno.h>
+
+#if __MP_LEGACY_SUPPORT_NEED_ATCALL_MACROS__
+#include <sys/fcntl.h>
+#include <fcntl.h>
+#include <unistd.h>
+#endif /* __MP_LEGACY_SUPPORT_NEED_ATCALL_MACROS__ */
+
+#define PROTECT_ERRNO(what)  ({ int __err = (errno); what; errno = __err; })
+#define ERR_ON(code, what)   if (what) { errno = (code); return -1; }
+
+#ifndef SYS___pthread_fchdir
+# define SYS___pthread_fchdir 349
+#endif
+
```

Comment:
extra \n to be removed

Reply:
That, likewise, is intentional. I like to group logical things via double newlines and find that the visual cues make reading code a bit easier. Personal preference, though.

If I remove it, I'd better also remove the other double newline for consistency.

Reply:
Hmmm ... I don't think we're consistent in this spacing. How about we just leave it as you've done it, since that's less work :)